### ISSUE TITLE
`sys.stdout` not detected as `TextIOBase`

### ISSUE THREAD

**Issue by qgallouedec** [python/mypy#15191]
**Bug Report**

See title.

**To Reproduce**

```python
from io import TextIOBase
import sys

def func(x: TextIOBase):
    pass

func(sys.stdout)
```

**Expected Behavior**

No error

**Actual Behavior**

```
mypy file.py
file.py:7: error: Argument 1 to "func" has incompatible type "TextIO"; expected "TextIOBase"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.2.0
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.10

**Comment by AlexWaygood** [comment]
This is a typeshed issue rather than a mypy issue. But this is a deliberate decision. For the reasons why we use these types for stdout, stdin and stderr (and how to workaround the error you're seeing in your code), please see the comment we recently added to the stub in https://github.com/python/typeshed/pull/10135/files.

See also recent discussion of this issue in https://github.com/python/typeshed/issues/10093
